port/unix/omrsysinfo.c

```diff
@@ -531,6 +576,810 @@ omrsysinfo_get_CPU_architecture(struct OMRPortLibrary *portLibrary)
 #endif
 }
 
+intptr_t
+omrsysinfo_get_processor_description(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc)
+{
+	intptr_t rc = -1;
+	Trc_PRT_sysinfo_get_processor_description_Entered(desc);
+
+	if (NULL != desc) {
+		memset(desc, 0, sizeof(OMRProcessorDesc));
+
+#if (defined(J9X86) || defined(J9HAMMER))
+		rc = getX86Description(portLibrary, desc);
+#elif defined(LINUXPPC)
+		rc = getLinuxPPCDescription(portLibrary, desc);
+#elif defined(AIXPPC)
+		rc = getAIXPPCDescription(portLibrary, desc);
+#elif (defined(S390) || defined(J9ZOS390))
+		rc = getS390Description(portLibrary, desc);
+#endif
+	}
+
+	Trc_PRT_sysinfo_get_processor_description_Exit(rc);
+	return rc;
+}
+
+BOOLEAN
+omrsysinfo_processor_has_feature(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc, uint32_t feature)
+{
+	BOOLEAN rc = FALSE;
+	Trc_PRT_sysinfo_processor_has_feature_Entered(desc, feature);
+
+#if defined(J9OS_I5)
+#if defined(J9OS_I5_V5R4)
+	if ((OMRPORT_PPC_FEATURE_HAS_VSX == feature) || (OMRPORT_PPC_FEATURE_HAS_ALTIVEC == feature) || (OMRPORT_PPC_FEATURE_HTM == feature)) {
+		Trc_PRT_sysinfo_processor_has_feature_Exit((UDATA)rc);
+		return rc;
+	}
+#elif defined(J9OS_I5_V6R1) || defined(J9OS_I5_V7R2)
+	if (J9PORT_PPC_FEATURE_HTM == feature) {
+		Trc_PRT_sysinfo_processor_has_feature_Exit((UDATA)rc);
+		return rc;
+	}
+#endif
+#endif
+
+	if ((NULL != desc)
+	&& (feature < (OMRPORT_SYSINFO_FEATURES_SIZE * 32))
+	) {
+		uint32_t featureIndex = feature / 32;
+		uint32_t featureShift = feature % 32;
+
+		rc = OMR_ARE_ALL_BITS_SET(desc->features[featureIndex], 1 << featureShift);
+	}
+
+	Trc_PRT_sysinfo_processor_has_feature_Exit((uintptr_t)rc);
+	return rc;
+}
+
+#if (defined(AIXPPC) || defined(S390) || defined(J9ZOS390))
+/**
+ * @internal
+ * Helper to set appropriate feature field in a OMRProcessorDesc struct.
+ *
+ * @param[in] desc pointer to the struct that contains the CPU type and features.
+ * @param[in] feature to set
+ *
+ */
+static void
+setFeature(OMRProcessorDesc *desc, uint32_t feature)
+{
+	if ((NULL != desc)
+	&& (feature < (OMRPORT_SYSINFO_FEATURES_SIZE * 32))
+	) {
+		uint32_t featureIndex = feature / 32;
+		uint32_t featureShift = feature % 32;
+
+		desc->features[featureIndex] = (desc->features[featureIndex] | (1 << (featureShift)));
+	}
+}
+#endif /* defined(AIXPPC) || defined(S390) || defined(J9ZOS390) */
+
+#if defined(LINUXPPC)
+/**
+ * @internal
+ * Populates J9ProcessorDesc *desc on Linux PPC
+ *
+ * @param[in] desc pointer to the struct that will contain the CPU type and features.
+ *
+ * @return 0 on success, -1 on failure
+ */
+static intptr_t
+getLinuxPPCDescription(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc)
+{
+	char* platform = NULL;
+	char* base_platform = NULL;
+
+	/* initialize auxv prior to querying the auxv */
+	if (prefetch_auxv()) {
```

Comment:
For clarity, this should be `!= 0`, because this function returns an error code, not a boolean.